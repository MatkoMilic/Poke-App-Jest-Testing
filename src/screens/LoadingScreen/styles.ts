import {StyleSheet} from 'react-native';

export default StyleSheet.create({
  titleView: {
    flex: 1,
    padding: 24,
    flexDirection: 'row',
    justifyContent: 'center',
  },
  titleText: {
    padding: 10,
  },
});
