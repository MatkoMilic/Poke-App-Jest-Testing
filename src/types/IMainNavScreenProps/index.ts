export type {
  IMainNavScreenProps,
  MainNavigationType,
} from './IMainNavScreenProps';
