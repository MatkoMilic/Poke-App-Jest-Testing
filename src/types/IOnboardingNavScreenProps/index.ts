export type {
  IOnboardingNavScreenProps,
  OnboardingNavigationType,
} from './IOnboardingNavScreenProps';
