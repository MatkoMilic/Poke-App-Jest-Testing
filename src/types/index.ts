export type {
  IMainNavScreenProps,
  MainNavigationType,
} from './IMainNavScreenProps';
export type {
  IOnboardingNavScreenProps,
  OnboardingNavigationType,
} from './IOnboardingNavScreenProps';
export type {
  MainNavigatorKeys,
  NavigatorNameKeys,
  OnboardingNavigatorParamList,
  RootNavigatorKeys,
  RootNavigatorParamList,
  OnboardingNavigatorKeys,
  MainNavigatorParamList,
  RootNavigatorType,
} from './navigatorTypes';
export {
  NavigatorNames,
  OnboardingNavigatorScreens,
  MainNavigatorScreens,
} from './navigatorTypes';
export type {IUserValues} from './IUserValues/IUserValues';
export {initialUserLoginValues} from './IUserValues/IUserValues';
