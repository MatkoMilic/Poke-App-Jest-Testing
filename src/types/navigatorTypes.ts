import {NavigatorScreenParams} from '@react-navigation/native';

export enum OnboardingNavigatorScreens {
  LOADING_SCREEN = 'LoadingScreen',
  LOGIN_SCREEN = 'LoginScreen',
}
export type OnboardingNavigatorParamList = {
  LOADING_SCREEN: undefined;
  LOGIN_SCREEN: undefined;
};
export type OnboardingNavigatorKeys = keyof OnboardingNavigatorScreens;
export enum MainNavigatorScreens {
  PROFILE_SCREEN = 'ProfileScreen',
  POKELIST_SCREEN = 'PokeListScreen',
  SETTINGS_SCREEN = 'SettingsScreen',
}
export type MainNavigatorParamList = {
  PROFILE_SCREEN: undefined;
  POKELIST_SCREEN: undefined;
  SETTINGS_SCREEN: undefined;
};
export type MainNavigatorKeys = keyof MainNavigatorScreens;
export enum NavigatorNames {
  MainNavigator = 'MainNavigator',
  OnboardingNavigator = 'OnboardingNavigator',
}
export type NavigatorNameKeys = keyof NavigatorNames;
export type RootNavigatorType = {
  MainNavigator: NavigatorScreenParams<MainNavigatorParamList>;
  OnboardingNavigator: NavigatorScreenParams<OnboardingNavigatorParamList>;
};
export type RootNavigatorParamList = {
  MainNavigator: undefined;
  OnboardingNavigator: undefined;
};
export type RootNavigatorKeys = keyof RootNavigatorType;
