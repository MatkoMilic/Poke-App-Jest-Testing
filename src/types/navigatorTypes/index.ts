export type {
  MainNavigatorKeys,
  OnboardingNavigatorKeys,
  MainNavigatorParamList,
  RootNavigatorParamList,
  RootNavigatorType,
  RootNavigatorKeys,
  NavigatorNameKeys,
  OnboardingNavigatorParamList,
} from './navigatorTypes';
export {
  NavigatorNames,
  OnboardingNavigatorScreens,
  MainNavigatorScreens,
} from './navigatorTypes';
