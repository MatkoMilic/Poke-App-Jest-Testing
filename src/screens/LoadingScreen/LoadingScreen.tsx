import React, {useEffect} from 'react';
import {Text, View} from 'react-native';
import {useTheme} from 'react-native-paper';
import AsyncStorage from '@react-native-async-storage/async-storage';
import styles from './styles';
import {StatusBar, ScreenContainer} from '../../components';
import {
  IOnboardingNavScreenProps,
  MainNavigatorScreens,
  NavigatorNames,
  OnboardingNavigatorScreens,
} from '../../types';

interface LoadingScreenProps extends IOnboardingNavScreenProps {}

const LoadingScreen: React.FC<LoadingScreenProps> = ({navigation}) => {
  const theme = useTheme();

  const chooseNavigator = async () => {
    const isUserLoggedIn = await AsyncStorage.getItem('activeUser');
    if (isUserLoggedIn !== null) {
      navigation.replace(NavigatorNames.MAIN_NAVIGATOR, {
        screen: MainNavigatorScreens.PROFILE_SCREEN,
      });
    } else {
      navigation.replace(OnboardingNavigatorScreens.LOGIN_SCREEN);
    }
  };

  useEffect(() => {
    chooseNavigator();
  }, []);

  return (
    <ScreenContainer>
      <StatusBar />
      <View style={styles.titleView}>
        <Text style={[{color: theme.colors.text}, styles.titleText]}>
          Loading...please wait
        </Text>
      </View>
    </ScreenContainer>
  );
};

export default LoadingScreen;
