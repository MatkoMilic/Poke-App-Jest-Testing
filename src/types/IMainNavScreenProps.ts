import {CompositeNavigationProp} from '@react-navigation/native';
import {NativeStackNavigationProp} from '@react-navigation/native-stack';
import {RootNavigatorType, MainNavigatorParamList} from './navigatorTypes';

export type MainNavigationType = CompositeNavigationProp<
  NativeStackNavigationProp<MainNavigatorParamList>,
  NativeStackNavigationProp<RootNavigatorType>
>;
export interface IMainNavScreenProps {
  navigation: MainNavigationType;
}
