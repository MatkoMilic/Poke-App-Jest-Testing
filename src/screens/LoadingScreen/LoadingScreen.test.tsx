import React from 'react';
import renderer from 'react-test-renderer';
import {OnboardingNavigationType} from '../../types';
import LoadingScreen from './LoadingScreen';

let navigation: Partial<OnboardingNavigationType>;
beforeEach(() => {
  navigation = {
    dispatch: jest.fn(),
  };
});
test('loading screen renders correctly', () => {
  const tree = renderer
    .create(
      <LoadingScreen navigation={navigation as OnboardingNavigationType} />,
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});
