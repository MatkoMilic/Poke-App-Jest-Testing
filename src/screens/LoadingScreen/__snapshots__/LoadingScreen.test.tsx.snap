// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`loading screen renders correctly 1`] = `
<RNCSafeAreaView
  style={
    Array [
      Object {
        "flex": 1,
      },
      undefined,
    ]
  }
>
  <View
    style={
      Object {
        "flex": 1,
        "flexDirection": "row",
        "justifyContent": "center",
        "padding": 24,
      }
    }
  >
    <Text
      style={
        Array [
          Object {
            "color": "#000000",
          },
          Object {
            "padding": 10,
          },
        ]
      }
    >
      Loading...please wait
    </Text>
  </View>
</RNCSafeAreaView>
`;
